subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -273,7 +275,10 @@ uint8_t ll_adv_params_set(uint16_t interval, uint8_t adv_type,
 	/* update the "current" primary adv data */
 	pdu = lll_adv_data_peek(&adv->lll);
 #if defined(CONFIG_BT_CTLR_ADV_EXT)
-	if (pdu->type != pdu_adv_type[adv_type]) {
+	if (is_new_set) {
+		pdu->type = pdu_adv_type[adv_type];
+		is_pdu_type_changed = 1;
+	} else if (!is_new_set && (pdu->type != pdu_adv_type[adv_type])) {
```

Comment:
`!is_new_set` is redundant here (`if (is_new_set) ... else`) and should be removed.

Reply:
@wopu-ot fixed